.github/workflows/integration.yaml

```diff
@@ -163,37 +163,28 @@ jobs:
           minikube image load docker.io/kubeflow/spark-operator:local
           ct install
 
-  integration-test:
-    runs-on: ubuntu-22.04
+  e2e-test:
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout source code
         uses: actions/checkout@v4
         with:
-          fetch-depth: "0"
+          fetch-depth: 0
 
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version-file: "go.mod"
+          go-version-file: go.mod
 
-      - name: setup minikube
-        uses: manusa/actions-setup-minikube@v2.11.0
+      - name: Create a Kind cluster
+        uses: helm/kind-action@v1
         with:
-          minikube version: v1.33.0
-          kubernetes version: v1.30.0
-          start args: --memory 6g --cpus=2 --addons ingress
-          github token: ${{ inputs.github-token }}
+          cluster_name: spark-operator
 
-      - name: Build local spark-operator docker image for minikube testing
+      - name: Build and load image to Kind cluster
         run: |
           docker build -t docker.io/kubeflow/spark-operator:local .
```

Comment:
We can utilize the `docker-build` target in the Makefile:

```bash
make docker-build IMAGE_TAG=local
```

Reply:
Changed this step to run `make kind-load-image` which also runs the `docker-build` target